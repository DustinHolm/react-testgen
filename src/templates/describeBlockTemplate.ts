export const createDescribeBlock = (name: string): string => {

    return `\
describe("${name} Tests", () => {
    beforeEach(() => {
        resetMocks()
    })

    test("Component renders", () => {
        whenComponentIsRendered()
        screen.debug()
    })
})
`
}