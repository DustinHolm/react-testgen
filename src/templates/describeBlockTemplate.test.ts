import {createDescribeBlock} from "./describeBlockTemplate"

describe("describeBlockTemplate Tests", () => {
    test("With name", () => {
        const returnedString = createDescribeBlock("test")
        expect(returnedString).toMatchSnapshot()
    })
})