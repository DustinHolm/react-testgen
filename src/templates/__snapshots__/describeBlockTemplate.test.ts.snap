// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describeBlockTemplate Tests With name 1`] = `
"describe(\\"test Tests\\", () => {
    beforeEach(() => {
        resetMocks()
    })

    test(\\"Component renders\\", () => {
        whenComponentIsRendered()
        screen.debug()
    })
})
"
`;
